klog_test.go

```diff
@@ -1852,3 +1857,48 @@ func (s *structWithLock) addWithDefer() {
 	defer s.m.Unlock()
 	s.n++
 }
+
+func TestFlushDaemon(t *testing.T) {
+	for sev := severity.InfoLog; sev < severity.FatalLog; sev++ {
+		flushed := make(chan struct{}, 1)
+		spyFunc := func() {
+			flushed <- struct{}{}
+		}
+		testClock := testingclock.NewFakeClock(time.Now())
+		testLog := loggingT{
+			flushD: &flushDaemon{
+				interval: time.Millisecond,
+				flush:    spyFunc,
+				clock:    testClock,
+			},
+		}
+
+		// Calling testLog will call createFile, which should start the daemon.
+		testLog.print(sev, nil, nil, "x")
+
+		if !testLog.flushD.isRunning() {
+			t.Error("expected flushD to be running")
+		}
+
+		testClock.Step(time.Second)
+		<-flushed
```

Comment:
Should this be verifying something?  What happens if there was a bug and flush-daemon did not call the func properly, what would this test report?

Reply:
The statement verifies that flush was called and let's the test wait until the flush execution finished. If the function isn't called, the test will time out. If I don't wait for the function here, the `testClock.Step()` statements will be all executed directly after each other, without flushDaemon having time to flush in between, so the flushDaemon would only enter the `case <-ticker.C()` once, regardless of how many steps the clock does.

Reply:
What I meant is maybe we should control the timeout and produce an error ourselves.  The default is something like 10 minutes, right?  It seems like we could wrap these reads in a timeout of 10 seconds and produce a better error.